.circleci: Prefer using env-file for docker run

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43293 .circleci: Prefer using env-file for docker run**

'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

seemethere committed:
.circleci: Prefer using env-file for docker run

'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere commented on .circleci/scripts/setup_ci_environment.sh:
```diff
@@ -47,31 +47,42 @@ else
   sudo rm -rf /etc/apt/sources.list.d/nvidia-docker.list
 fi
 
+add_to_env_file() {
```
Moving this to a function makes it easier to refactor if we decide to change where we want to store our `env-file`

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere commented:
This is now ready for review

ezyang reviewed (approved):
The change looks good at high level. If you want me to audit that you didn't "drop" any env vars, just shout; otherwise, trusting CI to tell you if something has gone missing.

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Prefer using env-file for docker run"


'docker run' has the capability to use a file for environment variables,
we should prefer to use that instead of having it be sourced per command
in the docker container.

Also opens the door for cutting down on the total number of commands we
need to echo into a script to then execute as a 'docker exec' command.

Plus side of this approach is that the BASH_ENV is persisted through all
of the steps so there's no need to do any exports / worry about
environment variables not persisting through jobs.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D23227059](https://our.internmc.facebook.com/intern/diff/D23227059)

[ghstack-poisoned]
